### ISSUE TITLE
Iteration order of linked maps are not in insertion order

### ISSUE THREAD

**Issue by sgjesse** [vigna/fastutil#83]
The following code does not print `ABCDE` as expected, but `BDCEA`.
```
    Object2IntMap<String> map = new Object2IntLinkedOpenHashMap<>();
    map.put("A", 1);
    map.put("B", 1);
    map.put("C", 1);
    map.put("D", 1);
    map.put("E", 1);
    map.keySet().forEach(System.out::print);
```

In fastutil version 7.2.0 the iteration order was as expected as is it for the following code using `java.util.LinkedHashMap`.

```
    Map<String, Integer> map = new LinkedHashMap<>();
    map.put("A", 1);
    map.put("B", 1);
    map.put("C", 1);
    map.put("D", 1);
    map.put("E", 1);
    map.keySet().forEach(System.out::print);
```

Might be caused by this commit: https://github.com/vigna/fastutil/commit/16abd16b3aeb655968f810d5a466ddde11487e29.

**Comment by sgjesse** [comment]
Changing from using `forEach` to `iterator().forEachRemaining` works around the issue.

**Comment by mikeb01** [comment]
Hopefully we can get this addressed.  A user (e.g. me) would expect .forEach() and .iterator() to have the same iteration order.

**Comment by vigna** [comment]
That was entirely stupid. I should have #ifdef'd a different version for linked maps. My apologies.

**Comment by simlu** [comment]
Just a FYI: I ran into this a while back and though I totally misunderstood what it was supposed to do. So abandoned and used a different data structure :) Good to see this is getting fixed now!

**Comment by vigna** [comment]
Fixed in 8.1.1.
